### ISSUE TITLE
Error when update addons

### ISSUE THREAD

**Issue by YorigamiShion** [layday/instawow#23]
Use `instawow update curse:details` and report 'instawow encountered an error'.

The error log:

```
2019-09-15 18:56:17.401 | ERROR    | instawow.manager:_intercept:385 - internal error
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\Scripts\instawow-script.py", line 11, in <module>
    load_entry_point('instawow==1.4.0', 'console_scripts', 'instawow')()
    �� <function load_entry_point at 0x000002EDDC9100D0>
  File "C:\ProgramData\Anaconda3\lib\site-packages\click\core.py", line 764, in __call__
    return self.main(*args, **kwargs)
           ��    ��     ��       �� {}
           ��    ��     �� ()
           ��    �� <function BaseCommand.main at 0x000002EDDCF302F0>
           �� <instawow.cli._OrigCmdOrderGroup object at 0x000002EDDE1972E8>
  File "C:\ProgramData\Anaconda3\lib\site-packages\click\core.py", line 717, in main
    rv = self.invoke(ctx)
         ��    ��      �� <click.core.Context object at 0x000002EDDCDDCC18>
         ��    �� <function MultiCommand.invoke at 0x000002EDDCF32048>
         �� <instawow.cli._OrigCmdOrderGroup object at 0x000002EDDE1972E8>
  File "C:\ProgramData\Anaconda3\lib\site-packages\click\core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
           ��               ��       ��       ��      �� <click.core.Context object at 0x000002EDDCE33FD0>
           ��               ��       ��       �� <function Command.invoke at 0x000002EDDCF30BF8>
           ��               ��       �� <click.core.Command object at 0x000002EDDE4BD080>
           ��               �� <click.core.Context object at 0x000002EDDCE33FD0>
           �� <function MultiCommand.invoke.<locals>._process_result at 0x000002EDDA3CC1E0>
  File "C:\ProgramData\Anaconda3\lib\site-packages\click\core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ��   ��      ��    ��           ��   �� {'addons': [('curse:details', ('curse', 'details'))]}
           ��   ��      ��    ��           �� <click.core.Context object at 0x000002EDDCE33FD0>
           ��   ��      ��    �� <function update at 0x000002EDDE4B6A60>
           ��   ��      �� <click.core.Command object at 0x000002EDDE4BD080>
           ��   �� <function Context.invoke at 0x000002EDDCF2FE18>
           �� <click.core.Context object at 0x000002EDDCE33FD0>
  File "C:\ProgramData\Anaconda3\lib\site-packages\click\core.py", line 555, in invoke
    return callback(*args, **kwargs)
           ��         ��       �� {'addons': [('curse:details', ('curse', 'details'))]}
           ��         �� ()
           �� <function update at 0x000002EDDE4B6A60>
  File "C:\ProgramData\Anaconda3\lib\site-packages\instawow-1.4.0-py3.7.egg\instawow\cli.py", line 136, in new_func
    return f(click.get_current_context().obj.m, *args, **kwargs)
           �� ��     ��                             ��       �� {'addons': [('curse:details', ('curse', 'details'))]}
           �� ��     ��                             �� ()
           �� ��     �� <function get_current_context at 0x000002EDDCF02598>
           �� �� <module 'click' from 'C:\\ProgramData\\Anaconda3\\lib\\site-packages\\click\\__init__.py'>
           �� <function update at 0x000002EDDE4B69D8>
  File "C:\ProgramData\Anaconda3\lib\site-packages\instawow-1.4.0-py3.7.egg\instawow\cli.py", line 200, in update
    results = zip((a for a, _ in values), manager.update(p for _, p in values))
                                 ��        ��       ��                    �� [('curse:details', ('curse', 'details'))]
                                 ��        ��       �� <property object at 0x000002EDDE4A96D8>
                                 ��        �� <instawow.manager.CliManager object at 0x000002EDDE521208>
                                 �� [('curse:details', ('curse', 'details'))]
  File "C:\ProgramData\Anaconda3\lib\site-packages\instawow-1.4.0-py3.7.egg\instawow\manager.py", line 405, in _process
    return self.run(do_process())
           ��    ��   �� <function CliManager._process.<locals>.do_process at 0x000002EDDE564D08>
           ��    �� <function Manager.run at 0x000002EDDE4B2400>
           �� <instawow.manager.CliManager object at 0x000002EDDE521208>
  File "C:\ProgramData\Anaconda3\lib\site-packages\instawow-1.4.0-py3.7.egg\instawow\manager.py", line 212, in run
    return cv.copy_context().run(runner)
           ��  ��                  �� <function Manager.run.<locals>.runner at 0x000002EDDE959D90>
           ��  �� <built-in function copy_context>
           �� <module 'contextvars' from 'C:\\ProgramData\\Anaconda3\\lib\\contextvars.py'>
  File "C:\ProgramData\Anaconda3\lib\site-packages\instawow-1.4.0-py3.7.egg\instawow\manager.py", line 210, in runner
    return loop.run_until_complete(do_run())
           ��    ��                  �� <function Manager.run.<locals>.runner.<locals>.do_run at 0x000002EDDE959E18>
           ��    �� <function BaseEventLoop.run_until_complete at 0x000002EDDD231620>
           �� <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\ProgramData\Anaconda3\lib\asyncio\base_events.py", line 555, in run_until_complete
    self.run_forever()
    ��    �� <function BaseEventLoop.run_forever at 0x000002EDDD231598>
    �� <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\ProgramData\Anaconda3\lib\asyncio\base_events.py", line 523, in run_forever
    self._run_once()
    ��    �� <function BaseEventLoop._run_once at 0x000002EDDD233F28>
    �� <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\ProgramData\Anaconda3\lib\asyncio\base_events.py", line 1758, in _run_once
    handle._run()
    ��      �� <function Handle._run at 0x000002EDDD1D8378>
    �� <Handle <TaskWakeupMethWrapper object at 0x000002EDDEDE6A38>(<_GatheringFu...column 65')}]>)>
  File "C:\ProgramData\Anaconda3\lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
    ��    ��            ��    ��           ��    �� <member '_args' of 'Handle' objects>
    ��    ��            ��    ��           �� <Handle <TaskWakeupMethWrapper object at 0x000002EDDEDE6A38>(<_GatheringFu...column 65')}]>)>
    ��    ��            ��    �� <member '_callback' of 'Handle' objects>
    ��    ��            �� <Handle <TaskWakeupMethWrapper object at 0x000002EDDEDE6A38>(<_GatheringFu...column 65')}]>)>
    ��    �� <member '_context' of 'Handle' objects>
    �� <Handle <TaskWakeupMethWrapper object at 0x000002EDDEDE6A38>(<_GatheringFu...column 65')}]>)>
  File "C:\ProgramData\Anaconda3\lib\site-packages\instawow-1.4.0-py3.7.egg\instawow\manager.py", line 206, in do_run
    return await awaitable
                 �� <coroutine object CliManager._process.<locals>.do_process at 0x000002EDDE8D43C8>
  File "C:\ProgramData\Anaconda3\lib\site-packages\instawow-1.4.0-py3.7.egg\instawow\manager.py", line 402, in do_process
    return [await _intercept(c) for c in coros.values()]
                  ��                      ��     �� <method 'values' of 'dict' objects>
                  ��                      �� {('curse', 'details'): <coroutine object _error_wrapper.<locals>.error_out at 0x000002EDDE8D4948>}
                  �� <function _intercept at 0x000002EDDE4B29D8>
  File "C:\ProgramData\Anaconda3\lib\site-packages\instawow-1.4.0-py3.7.egg\instawow\manager.py", line 402, in <listcomp>
    return [await _intercept(c) for c in coros.values()]
                  ��          ��      �� <coroutine object _error_wrapper.<locals>.error_out at 0x000002EDDE8D4948>
                  ��          �� <coroutine object _error_wrapper.<locals>.error_out at 0x000002EDDE8D4948>
                  �� <function _intercept at 0x000002EDDE4B29D8>
> File "C:\ProgramData\Anaconda3\lib\site-packages\instawow-1.4.0-py3.7.egg\instawow\manager.py", line 381, in _intercept
    return await coro
                 �� <coroutine object _error_wrapper.<locals>.error_out at 0x000002EDDE8D4948>
  File "C:\ProgramData\Anaconda3\lib\site-packages\instawow-1.4.0-py3.7.egg\instawow\manager.py", line 179, in error_out
    raise error
          �� ParseError('undefined entity: line 78, column 65')
  File "C:\ProgramData\Anaconda3\lib\site-packages\instawow-1.4.0-py3.7.egg\instawow\manager.py", line 224, in resolve
    return await self.resolvers[source].resolve(values, strategy=strategy)
                 ��    ��         ��               ��       ��        �� 'default'
                 ��    ��         ��               ��       �� 'default'
                 ��    ��         ��               �� ['details']
                 ��    ��         �� 'curse'
                 ��    �� {'tukui': <instawow.resolvers.TukuiResolver object at 0x000002EDDE4BDA58>, 'curse': <instawow.resolvers.CurseResolver object ...
                 �� <instawow.manager.CliManager object at 0x000002EDDE521208>
  File "C:\ProgramData\Anaconda3\lib\site-packages\instawow-1.4.0-py3.7.egg\instawow\resolvers.py", line 130, in resolve
    for k, v in (await self._fetch(ids)).items())
                       ��    ��      �� ['details']
                       ��    �� <function CurseResolver._fetch at 0x000002EDDE4AC488>
                       �� <instawow.resolvers.CurseResolver object at 0x000002EDDE4BD8D0>
  File "C:\ProgramData\Anaconda3\lib\site-packages\instawow-1.4.0-py3.7.egg\instawow\resolvers.py", line 109, in _fetch
    extracted = await gather((extract_id(i) for i in ids), False)
                      ��       ��                      �� ['details']
                      ��       �� <function CurseResolver._fetch.<locals>.extract_id at 0x000002EDDEE049D8>
                      �� <function gather at 0x000002EDDE1F8048>
  File "C:\ProgramData\Anaconda3\lib\site-packages\instawow-1.4.0-py3.7.egg\instawow\utils.py", line 116, in gather
    return await asyncio.gather(*it, return_exceptions=return_exceptions)
                 ��       ��       ��   ��                 �� False
                 ��       ��       ��   �� False
                 ��       ��       �� <generator object CurseResolver._fetch.<locals>.<genexpr> at 0x000002EDDED8EC78>
                 ��       �� <function gather at 0x000002EDDD22F7B8>
                 �� <module 'asyncio' from 'C:\\ProgramData\\Anaconda3\\lib\\asyncio\\__init__.py'>
  File "C:\ProgramData\Anaconda3\lib\site-packages\instawow-1.4.0-py3.7.egg\instawow\resolvers.py", line 105, in extract_id
    xml = ElementTree.fromstring(await response.text())
          ��           ��                ��        �� <function ClientResponse.text at 0x000002EDDED7B158>
          ��           ��                �� <ClientResponse(https://www.curseforge.com/wow/addons/details/download-client) [503 Service Temporarily Unavailable]>
          ��           ��                  <CIMult...
          ��           �� <function XML at 0x000002EDDEE6D488>
          �� <module 'xml.etree.ElementTree' from 'C:\\ProgramData\\Anaconda3\\lib\\xml\\etree\\ElementTree.py'>
  File "C:\ProgramData\Anaconda3\lib\xml\etree\ElementTree.py", line 1315, in XML
    parser.feed(text)
    ��      ��    �� '<!DOCTYPE HTML>\n<html lang="en-US">\n<head>\n  <meta charset="UTF-8" />\n  <meta http-equiv="Content-Type" content="text/ht...
    ��      �� <method 'feed' of 'xml.etree.ElementTree.XMLParser' objects>
    �� <xml.etree.ElementTree.XMLParser object at 0x000002EDDE8D4748>
  File "<string>", line None

xml.etree.ElementTree.ParseError: undefined entity: line 78, column 65
```

I am using a proxy and I can access `https://www.curseforge.com/wow/addons/details/download-client` when using proxy on broswer.

**Comment by layday** [comment]
CurseForge has been routing all of their traffic through Cloudflare.  Sometimes, *instawow* will trigger Cloudflare's DDoS interstitial.  The interstitial is designed to perform a JavaScript redirect after five seconds.  *instawow* doesn't support JavaScript - it does not wait for the redirect - and will attempt to parse the interstitial as if it were the response it was expecting.  Cloudflare can distinguish between your browser and *instawow*, which is why the page is accessible from your browser.  I suppose we could periodically scrape all the slugs from the API so we never have to deal with the CurseForge website in real time.

**Comment by layday** [comment]
Fixed in v1.4.2.
